Update fast/mediastream/microphone-interruption-and-audio-session.html to get more meaningful error messages

youennf opened the pull request:
#### c1aeb1763d76f2f2dc6ea4211c5ad75e6c25bd63
<pre>
Update fast/mediastream/microphone-interruption-and-audio-session.html to get more meaningful error messages
<a href="https://rdar.apple.com/138912337">rdar://138912337</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=282328">https://bugs.webkit.org/show_bug.cgi?id=282328</a>

Reviewed by Jean-Yves Avenard.

Listen to mute/unmute events and reject the promise after a timeout of 1 second to have the test fail early with an error message.

* LayoutTests/fast/mediastream/microphone-interruption-and-audio-session.html:

Canonical link: <a href="https://commits.webkit.org/285948@main">https://commits.webkit.org/285948@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5c3e30e4756c5233258c3baba0b8b32a93bdcfbb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74145 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53574 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26956 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78518 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25383 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62707 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1359 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58294 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16634 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77212 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48450 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63782 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38704 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45361 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21278 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23716 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66832 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21624 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80038 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1462 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/818 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66595 "Found 12 new test failures: compositing/animation/repaint-after-clearing-shared-backing.html imported/w3c/web-platform-tests/css/css-pseudo/marker-animate-002.html imported/w3c/web-platform-tests/css/css-view-transitions/navigation/root-element-transition-iframe-with-startVT-on-main.html imported/w3c/web-platform-tests/css/css-view-transitions/new-content-ancestor-clipped-2.html imported/w3c/web-platform-tests/css/css-view-transitions/new-content-captures-positioned-spans.html imported/w3c/web-platform-tests/css/css-view-transitions/old-content-captures-clip-path.html imported/w3c/web-platform-tests/css/css-view-transitions/old-content-object-view-box-overflow.html imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-element-preserve-3d.html imported/w3c/web-platform-tests/css/css-view-transitions/snapshot-containing-block-includes-scrollbar-gutter.html imported/w3c/web-platform-tests/css/css-view-transitions/span-with-overflowing-text-and-box-decorations.html ... (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1606 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63799 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65868 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16368 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9805 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7961 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1426 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1455 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1443 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1462 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on LayoutTests/fast/mediastream/microphone-interruption-and-audio-session.html:
```diff
@@ -8,6 +8,22 @@
  </head>
 <body>
     <script>
+    function waitForMute(track, testName)
```
would be nicer if the timeout value was provided as a parameter, or indicate in the name that it waits for a maximum of 1s

jyavenard commented on LayoutTests/fast/mediastream/microphone-interruption-and-audio-session.html:
```diff
@@ -8,6 +8,22 @@
+        });
+    }
+
+    function waitForUnmute(track, testName)
```
ditto

youennf committed:
Update fast/mediastream/microphone-interruption-and-audio-session.html to get more meaningful error messages
rdar://138912337
https://bugs.webkit.org/show_bug.cgi?id=282328

Reviewed by Jean-Yves Avenard.

Listen to mute/unmute events and reject the promise after a timeout of 1 second to have the test fail early with an error message.

* LayoutTests/fast/mediastream/microphone-interruption-and-audio-session.html:

Canonical link: https://commits.webkit.org/285948@main
